File: app/src/main/java/com/roy/pokemonbook/binding/ViewBinding.java
package com.roy.pokemonbook.binding;

import android.widget.ImageView;

import androidx.databinding.BindingAdapter;

import com.bumptech.glide.Glide;
import com.roy.pokemonbook.R;

public final class ViewBinding {

    //bindingAdapter必须是static的
    @BindingAdapter("bindingImage")
    public static void bindImage(ImageView imageView, String url) {
        Glide.with(imageView.getContext())
                .load(url)
                .placeholder(R.drawable.ic_pokemon_default)
                .into(imageView);
    }
}

File: app/src/main/java/com/roy/pokemonbook/binding/ViewBinding.kt
package com.roy.pokemonbook.binding


import android.widget.ImageView
import androidx.databinding.BindingAdapter
import coil.load
import com.roy.pokemonbook.R


@BindingAdapter("bindingAvator")
fun bindingAvator(imageView: ImageView, url: String) {
    imageView.load(url) {
        crossfade(true)
        placeholder(R.mipmap.ic_launcher_round)
    }
}
